addons/calendar/static/src/js/base_calendar.js

```diff
@@ -11,6 +16,170 @@ var WebClient = require('web.WebClient');
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * This method call _deleteRecurrentEvents with the type as the action selected by
+     * the user and the calendar event id.
+     * Once the event has been deleted, it will close the modal and refresh the view.
+     *
+     * @param {OdooEvent} event
+     */
+    _onDeleteRecurrentEvent: function (event){
+        var self = this;
+        var type = event.data.type;
+        var id = event.target.id;
+        Dialog.confirm(this, _t("Are you sure you want to delete this/these record(s) ?"), {
+            confirm_callback: function () {
+                return self.model.deleteRecurrentEvents(id, type).then(function () {
+                    self.dialog.destroy();
+                    self.reload();
+                });
+            }
+        });
+    },
+});
+
+CalendarModel.include({
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * This method call _splitRecurrentEvents with a different python method depending of the
+     * type. It then get the id of event to delete and call the super function deleteRecords that
+     * will unlink it in the database.
+     *
+     * @param {OdooEvent} type The event data type represent the action selected by the user.
+     * @param {OdooEvent} id Event id of original event.
+     */
+    deleteRecurrentEvents : function (id, type){
+        var self = this;
+        var method;
+        switch(type){
+            case 'one':
+                method = 'action_detach_recurring_event';
+                break;
+            case 'future':
+                method = 'action_future_recurring_event';
+                break;
+            case 'all':
+                method = 'action_all_recurring_event';
+                break;
+        }
+        return self._splitRecurrentEvents(id, method).then(function (r) {
+            self.deleteRecords(r.res_id, self.modelName);
+        });
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Call the server in order to detach the part of the event that we want to delete.
+     *
+     * @param {integer} id The selected calendar event id.
+     * @param {string} method A python method name in order to get the id of the events to remove
+     * the expected method are :
+     *    -action_detach_recurring_event to only remove the selected event.
+     *    -action_future_recurring_event to remove the selected event and the following.
+     *    -action_all_recurring_event to remove all events.
+     */
+    _splitRecurrentEvents : function (id, method){
```

Comment:
you can move rpc call inside deleteRecurrentEvents (and remove _splitRecurrentEvents)